soc/xtensa/intel_adsp/common/include/cavs_hda.h

```diff
@@ -0,0 +1,303 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_CAVS_HDA_H
+#define ZEPHYR_INCLUDE_CAVS_HDA_H
+
+#include <arch/xtensa/cache.h>
+#include <kernel.h>
+#include <device.h>
+#include <cavs-shim.h>
+#include <cavs-mem.h>
+
+/**
+ * @brief HDA stream functionality for cAVS
+ *
+ * Provides low level calls to support cAVS HDA streams with
+ * minimal abstraction that allows testing the hardware
+ * and its demands separately from the intended DMA API
+ * usage. The only requirement is that you define the base
+ * addresses, the stream count, and the size of the ip blocks.
+ */
+
+#ifndef HDA_HOST_OUT_BASE
+#error Must define HDA_HOST_OUT_BASE
+#endif
+#ifndef HDA_HOST_IN_BASE
+#error Must define HDA_HOST_IN_BASE
+#endif
+#ifndef HDA_STREAM_COUNT
+#error Must define HDA_STREAM_COUNT
+#endif
+#ifndef HDA_REGBLOCK_SIZE
+#error Must define HDA_REGBLOCK_SIZE
+#endif
+
+/* The read/write positions are masked to 24 bits */
+#define HDA_RWP_MASK 0x00FFFFFF
+
+/* Buffers must be 128 byte aligned, this mask enforces that */
+#define HDA_ALIGN_MASK 0xFFFFFF80
+
+
+#define HDA_ADDR(base, stream) ((base) + (stream)*HDA_REGBLOCK_SIZE)
+
+
+/* Gateway Control and Status Register */
+#define DGCS(base, stream) ((volatile uint32_t *)HDA_ADDR(base, stream))
+#define DGCS_SCS BIT(31) /* Sample container size */
+#define DGCS_GEN BIT(26) /* Gateway Enable */
+#define DGCS_L1ETP BIT(25) /* L1 Enter Prevent */
+#define DGCS_L1EXP BIT(25) /* L1 Exit Prevent */
+#define DGCS_FWCB BIT(23) /* Firmware Control Buffer */
+#define DGCS_GBUSY BIT(15) /* Gateway Busy */
+#define DGCS_TE BIT(14) /* Transfer Error */
+#define DGCS_BSC BIT(11) /* Buffer Segment Completion */
+#define DGCS_BOR BIT(10) /* Buffer Overrun */
+#define DGCS_BUR BIT(10) /* Buffer Underrun */
+#define DGCS_BF BIT(9) /* Buffer Full */
+#define DGCS_BNE BIT(8) /* Buffer Not Empty */
+#define DGCS_FIFORDY BIT(5) /* Enable FIFO */
+#define DGCS_BSCIE BIT(3) /* Buffer Segment Completion Interrupt Enable */
+
+/* Gateway Buffer Base Address */
+#define DGBBA(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x04))
+
+/* Gateway Buffer Size */
+#define DGBS(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x08))
+
+/* Gateway Buffer Position Increment */
+#define DGBFPI(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x0c))
+
+/* Gateway Buffer Read Position */
+#define DGBRP(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x10))
+
+/* Gateway Buffer Write Position */
+#define DGBWP(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x14))
+
+/* Gateway Buffer Segment Position */
+#define DGBSP(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x18))
+
+/* Gateway Minimum Buffer Size */
+#define DGMBS(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x1c))
```

Comment:
I was checking registry values in real audio playback scenario and see no use of DGMBS, this one is the only difference (SOF will put buffer size here).
Not a big deal, probably doesnt matter, just FYI